ERCS/erc-7715.md

```diff
@@ -0,0 +1,466 @@
+
+In the context of AA, there are multiple vendor-specific implementations of session keys, which are temporary keys imbued with specific permissions.  However, since the implementations are vendor-specific, it’s impossible for DApps to “request” session keys from wallets in a unified way, regardless of the specific wallet implementations.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### `wallet_issuePermissions`
+
+We introduce a `wallet_issuePermissions` method for the DApp to request permissions from the wallet.  We’d prefer `wallet_requestPermissions` but it’s already taken with [ERC-2255](./eip-2255.md).
+
+#### Request Specification
+
+```tsx
+type IssuePermissionsRequestParams = {
+  signer?: {
+    type: string
+    data: any
+  }
+ 
+  permissions: {
+    type: string
+    data: any
+    required: boolean
+  }[]
+  
+  expiry: number,
+}
+```
+
+`signer` is an optional field that identifies the key or account associated with the session.  When not specified, it’s implied that the wallet will manage the session.  See the “Wallet-managed Sessions” section for details.
+
+`permissions` is a list of permissions.  See “Permission Types” section for details.
+
+Each permission can be either `required` or not.  If a permission is `required`, the wallet MUST grant the permission if it approves the request.  Otherwise, the wallet may reject the permission even if it approves the request (presumably because it grants other permissions).
+
+`expiry` is a UNIX timestamp that specifies the time by which this session MUST expire.
+
+
+Request example:
+
+```tsx
+{
+  signer: {
+    type: 'key',
+    data: {
+      id:'did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme',
+      }
+    }
+  permissions: [
+    {
+      type: 'erc20_spending_limit',
+      data: {
+        erc20Address: '0x...',
+        limit: '0x...',
+      }
+      required: true
+    },
+    {
+      type: 'gas_limit',
+      data: {
+        limit: '0x...',
+      },
+      required: false
+    }
+  ],
+  expiry: 1577840461
+}
+```
+
+#### Response Specification
+
+```tsx
+
+type IssuePermissionsResponse = {
+  grantedPermissions: {
+    type: string
+    data: any
+  }[]
+      
+  expiry: number
+  
+  signerData: {
+    userOpBuilder?: `0x${string}`
+    submitToAddress?: `0x${string}`
+  },
+  factory?: `0x${string}`,
+  factoryData?: string,
+	
+  permissionsContext: string
+}
+```
+
+- `grantedPermissions` is a list of permissions that were granted.  The wallet MUST grant `required` permissions if the request was approved.
+- `expiry` is the expiry time that was granted.  The wallet MAY grant a different expiry time than the one that the DApp requested.
```

Comment:
can we describe about `signerData.userOpBuilder` and `signerData.submitToAddress` fields.